frc53_nft/src/lib.rs

```diff
@@ -412,6 +415,56 @@ where
         Ok(self.state.transfer_return(&self.runtime, intermediate)?)
     }
 
+    /// Enumerates a page of TokenIDs
+    pub fn list_tokens(&self, cursor: Option<AmtCursor>, max: u64) -> Result<ListTokensReturn> {
+        let (tokens, next_cursor) = self.state.list_tokens(&self.runtime, cursor, Some(max))?;
+        Ok(ListTokensReturn { tokens, next_cursor })
+    }
+
+    /// Enumerates a page of TokenIDs owned by a specific address
+    pub fn list_owned_tokens(
+        &self,
+        owner: &Address,
+        cursor: Option<AmtCursor>,
+        max: u64,
+    ) -> Result<ListTokensReturn> {
+        let owner_id = self.runtime.resolve_id(owner)?;
+        let (tokens, next_cursor) =
+            self.state.list_owned_tokens(&self.runtime, owner_id, cursor, Some(max))?;
+        Ok(ListTokensReturn { tokens, next_cursor })
+    }
+
+    /// Returns all the operators approved by an owner for a token
+    pub fn list_token_operators(&self, owner: &Address, token_id: TokenID) -> Result<Vec<ActorID>> {
+        let _owner_id = self.runtime.resolve_id(owner)?;
```

Comment:
not sure if there is value in asserting that the given token is owned by a specific address, I suggest we modify the FIP to not require this parameter.

else I'll add an additional check at the state-level functionality

Reply:
Great catch, thank you.